client/src/js/components/Bus.js

```diff
@@ -281,90 +319,151 @@ const createStatusComponent = (status, t) => {
-            </StyledSubHeading>
-            <Status>{wasedaStatusComponent.status}</Status>
-            <Remark>{wasedaStatusComponent.remark}</Remark>
-          </BusStatus>
-          <BusStatus>
-            <StyledSubHeading>
-              {t("bus.NishiWaseda")}{" "}
-              <FontAwesomeIcon icon={faAngleDoubleRight} size="1x" />{" "}
-              {t("bus.Waseda")}
-            </StyledSubHeading>
-            <Status>{nishiStatusComponent.status}</Status>
-            <Remark>{nishiStatusComponent.remark}</Remark>
-          </BusStatus>
-          <ModalContainer
-            linkText={t("bus.Add to home screen")}
-            text={t("bus.and never miss a bus again!")}
-          >
-            <ModalArticle>
-              <ModalSection>
-                <ModalHeading>Android / Chrome:</ModalHeading>
-                <p>
-                  {t("bus.Tap on the top-right icon")}
-                  &nbsp;
-                  <FontAwesomeIcon icon={faEllipsisV} size="1x" />
-                  &nbsp;
-                  {t("bus.and select Add to Home screen")}
-                </p>
-                <ModalImage
-                  src={a2hsChrome}
-                  alt="Add to home screen image for Chrome"
+const DateSelector = ({ value, onClick }) => (
+  <span onClick={onClick}>
+    <FontAwesomeIcon icon={faCalendarAlt} size="1x" />{" "}
+    {value}
+  </span>
+);
+
+const TimeSelector = ({ value, onClick }) => (
+  <span onClick={onClick}>
+    <FontAwesomeIcon icon={faClock} size="1x" />{" "}
+    {value}
+  </span>
+);
+
+class Bus extends React.Component {
+  state = {
+    date: new Date()
+  };
+
+  onDatetimeChange = date => {
+    this.setState({ date: date || new Date() })
+  }
+
+  clearDatetime = () => {
+    this.setState({ date: new Date() })
+  }
+
+  render () {
+    const { t, lng } = this.props;
+    const { wasedaStatus, nishiStatus } = getBusStatuses(this.state.date, lng, t);
+    const wasedaStatusComponent = createStatusComponent(wasedaStatus, t);
+    const nishiStatusComponent = createStatusComponent(nishiStatus, t);
+    return (
+      <Wrapper>
+        <Helmet>
+          <title>WasedaTime -　Bus</title>
+          <meta
+            name="description"
+            content="Shuttle Bus Arrival Time Checking at Waseda University."
+            />
+          <meta property="og:title" content="WasedaTime - Bus" />
+          <meta
+            property="og:description"
+            content="Shuttle Bus Arrival Time Checking at Waseda University."
+            />
+          <meta property="og:site_name" content="WasedaTime - Bus" />
+        </Helmet>
+        <ExtendedOverlay>
+          <InfoWrapper>
+            <StyledHeading>{t("bus.busStatus")}</StyledHeading>
+            <p>
+              <FontAwesomeIcon icon={faSearch} size="1x" />{' '}
+              {t("bus.Assign a date / time to check the next bus")}：
+            </p>
+
+            <DatetimeSelection>
+              <DatePickerWrapper>
```

Comment:
Let DatePickerWrapper (Don't forget to change it to a button!) handle onClick instead of DateSelector and TimeSelector.

If we handle onClick with a span, the calendar will open ONLY when we click on the black text. This means clicking on the red circle area doesn't do anything, which is not good.
![Screen Shot 2020-06-01 at 18 21 49](https://user-images.githubusercontent.com/30998659/83401025-900a3a00-a436-11ea-8e1e-bcb9916cf4cb.png)

Reply:
The onclick method only works in the component passed as the `customInput` prop, so I could not let `DatePickerWrapper` handle the onclick method.
Instead, I adjusted the composition of the buttons and their widths, and then the area doing nothing is successfully removed.